src/snowflake/connector/aio/_result_batch.py

```diff
@@ -191,14 +191,27 @@ async def create_iter(
 
     async def _download(
         self, connection: SnowflakeConnection | None = None, **kwargs
-    ) -> aiohttp.ClientResponse:
+    ) -> tuple[bytes, str]:
```

Comment:
just curious, what is the reason that we directly return downloaded data instead of response?

Reply:
good question, this is because we use `async with connection._rest._use_requests_session() as session:` when issuing the request.

the response object doesn't work if it goes out of the scope of this session -- it will not be able to download data out of the scope of the session